source/s3_chunk_stream.c

```diff
@@ -253,12 +255,13 @@ struct aws_input_stream *aws_chunk_stream_new(
     }
     /* we subtract one since aws_base64_compute_encoded_len accounts for the null terminator which won't show up in our
      * stream */
-    impl->length = prechunk_stream_len + stream_length + final_chunk_len + impl->checksum_header_name.len + colon_len +
+    impl->length = prechunk_stream_len + stream_length + final_chunk_len + impl->checksum_header_name->len + colon_len +
                    encoded_checksum_len + post_trailer_len - 1;
     AWS_FATAL_ASSERT(impl->current_stream);
     return stream;
 
 error:
+    /* TODO: the stream should refcount, otherwise, this can lead to crash */
     aws_input_stream_destroy(impl->checksum_stream);
```

Comment:
Feel like a bug.
So, if error from place after `impl->checksum_stream == NULL`, this will destroy the `existing_stream`, if error from place before that, it will NOT destroy the `existing_stream`.

The right thing should be if this function fails, it should NEVER destroy the `existing_stream`. Will be easier if we have refcount, but we can still do that to avoid the bug.